.build/build-sonar.xml

```diff
@@ -0,0 +1,167 @@
+            </retry>
+            <unzip src="${sonar.download.dir}/sonar-scanner-cli-${sonar-scanner-cli.version}.zip" dest="${sonar.download.dir}" overwrite="true"/>
+            <setpermissions mode="755">
+                <!-- unzip task in Ant does not preserve file permissions, so we need to make those files executable explicitly -->
+                <file file="${sonar-scanner.home}/bin/sonar-scanner" />
+                <file file="${sonar-scanner.home}/jre/bin/java" />
+                <file file="${sonar-scanner.home}/jre/lib/jspawnhelper" />
+            </setpermissions>
+        </sequential>
+    </target>
+
+    <macrodef name="check-server-status">
+        <attribute name="url"/>
+        <sequential>
+            <exec executable="bash" failifexecutionfails="true" failonerror="true" logerror="false">
+                <arg value="-c"/>
+                <arg value="curl -s &quot;@{url}/api/system/status&quot; | jq -r &quot;.status&quot; | grep UP"/>
+            </exec>
+        </sequential>
+    </macrodef>
+
+    <target name="init-sonar-server" depends="sonar-init,maybe-set-sonar-server-url,maybe-start-sonar-server">
+        <check-server-status url="${sonar.host.url}"/>
+    </target>
+
+    <target name="maybe-set-sonar-server-url" depends="sonar-init" if="env.SONAR_SERVER_URL">
+        <property name="sonar.host.url" value="${env.SONAR_SERVER_URL}"/>
+        <fail unless="env.SONAR_CASSANDRA_TOKEN" message="SONAR_CASSANDRA_TOKEN environment variable must be set when using Sonar server defined in env SONAR_SERVER_URL"/>
+        <property name="sonar.token" value="${env.SONAR_CASSANDRA_TOKEN}"/>
+    </target>
+
+    <target name="maybe-start-sonar-server" depends="sonar-init" unless="env.SONAR_SERVER_URL">
+        <property name="sonar.host.url" value="http://127.0.0.1:9000"/>
+        <local name="sonar-server.running"/>
+        <condition property="sonar-server.running" value="true">
+            <socket server="127.0.0.1" port="9000"/>
+        </condition>
+        <antcall target="start-sonarqube" unless:set="sonar-server.running" inheritrefs="true"/>
+
+        <exec executable="bash" failifexecutionfails="false" failonerror="true" logerror="false">
+            <arg line="-c"/>
+            <arg value="curl -s -u admin:password -X POST '${sonar.host.url}/api/user_tokens/revoke?name=cassandra'"/>
+        </exec>
+
+        <exec executable="bash" failifexecutionfails="true" failonerror="true" logerror="false">
+            <arg line="-c"/>
+            <arg value="curl -s -u admin:password -X POST '${sonar.host.url}/api/user_tokens/generate?name=cassandra&amp;type=GLOBAL_ANALYSIS_TOKEN' | jq -r '.token'"/>
+            <redirector outputproperty="sonar.token"/>
+        </exec>
+    </target>
+
+    <target name="start-sonarqube" depends="sonar-init">
+        <exec executable="docker" dir="${sonar.workdir}" failifexecutionfails="true" failonerror="true" logerror="true">
```

Comment:
The script that was created here: [In-tree build scripts](https://issues.apache.org/jira/browse/CASSANDRA-18133) runs and stops the docker image. Should we reuse this approach here as well? wdyt? And here we can just call the sh we need.

Reply:
We don't want to stop the server because we would not be able to browse the results.

Reply:
Can we report the prepared result to the console? While the docker is running I think we can fetch all the results from the API and push them to the console. 

The two things that concern me the most: 
- We leave unclosed resources after the build process. To me, the `ant` is a build tool, and not designed to be a server-like script when we initiate something to start like a daemon;
- Another point is if it leaves the resources unclosed, it will probably only be used on local machines, which, in turn, concerns why not just use simple `bash` instead, like in-tree scripts.

Reply:
I think this is out of the scope. We could probably export list of issues in JSON, but that is a bit unreadable without some processing.

Reply:
JSON report is saved in the build directory after analysis; in a separate ticket we can try to convert it to JUnit compatible report so that Jenkins can display it properly